pkg/controllers/rollout/controller.go

```diff
@@ -408,30 +411,73 @@ func (r *Reconciler) pickBindingsToRoll(ctx context.Context, allBindings []*flee
 			}
 		}
 	}
+	minWaitTime := time.Duration(0)
+	if len(waitTimes) > 0 {
+		// Sort the wait times in ascending order
+		sort.Slice(waitTimes, func(i, j int) bool {
+			return waitTimes[i] < waitTimes[j]
+		})
+		minWaitTime = waitTimes[0]
+	}
```

Comment:
find the min in an array is an O(N) time, O(1) space problem, this implementation is O(NlogN) time O(N) space.